### PR TITLE
Updates frontend to Typescript to 3.3.1

### PR DESCRIPTION
**Pull request by rileyjbauer**
Also updates code around `CustomTable customRenderer` to make better use of `FunctionComponent` typings.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/772)

### CODE REVIEW

File: frontend/src/pages/ExperimentList.test.tsx [hunk 1]
```diff
@@ -70,14 +71,22 @@ describe('ExperimentList', () => {
     formatDateStringSpy.mockImplementation(() => '1/2/2019, 12:34:56 PM');
   });
 
+  afterEach(async () => {
+    // unmount() should be called before resetAllMocks() in case any part of the unmount life cycle
+    // depends on mocks/spies
+    if (tree) {
```
Comment (yebrahim) [inline frontend/src/pages/ExperimentList.test.tsx]: You don't need the check. I don't think anything happens if the tree is unmounted.

Comment (yebrahim) [reply frontend/src/pages/ExperimentList.test.tsx]: Why the `await`?

File: frontend/src/pages/RunList.test.tsx [hunk 2]
```diff
@@ -83,14 +95,23 @@ describe('RunList', () => {
     getExperimentSpy.mockClear();
   });
 
+  afterEach(async () => {
+    // unmount() should be called before resetAllMocks() in case any part of the unmount life cycle
+    // depends on mocks/spies
+    if (tree) {
```
Comment (yebrahim) [inline frontend/src/pages/RunList.test.tsx]: Same as comment above.

File: frontend/src/pages/ExperimentList.test.tsx (hunk 1 above)
Comment (rileyjbauer) [reply frontend/src/pages/ExperimentList.test.tsx]: I was seeing errors trying to unmount the tree if it was already unmounted (or had never been mounted).

It seems like the `await` probably isn't needed. We do it in many other tests though. I'll go through and clean that up

**Pull request merged**